Gems/ONNX/Code/Source/Clients/Model.cpp

```diff
@@ -0,0 +1,153 @@
+            {
+                if (azInputShape[index] == -1)
+                {
+                    azInputShape[index] = 1;
+                }
+            }
+            m_inputShapes.push_back(azInputShape);
+        }
+
+        // Extract output names from model file and put into const char* vectors.
+        // Extract output shapes from model file and put into AZStd::vector<int64_t>.
+        // Initialize m_outputs vector using output shape and count.
+        m_outputCount = m_session.GetOutputCount();
+        AZStd::vector<AZStd::vector<float>> outputs(m_outputCount);
+        for (size_t i = 0; i < m_outputCount; i++)
+        {
+            const char* out_name = m_session.GetOutputName(i, *m_allocator);
+            m_outputNames.push_back(out_name);
+
+            std::vector<int64_t> outputShape = m_session.GetOutputTypeInfo(0).GetTensorTypeAndShapeInfo().GetShape();
+            AZStd::vector<int64_t> azOutputShape(outputShape.begin(), outputShape.end());
+            for (int index = 0; index < azOutputShape.size(); index++)
+            {
+                if (azOutputShape[index] == -1)
+                {
+                    azOutputShape[index] = 1;
+                }
+            }
+            m_outputShapes.push_back(azOutputShape);
+
+            int64_t outputSize = 1;
+            for (int j = 0; j < m_outputShapes[i].size(); j++)
+            {
+                // The size of each output is simply all the magnitudes of the shape dimensions multiplied together.
+                if (m_outputShapes[i][j] > 0)
+                {
+                    outputSize *= m_outputShapes[i][j];
+                }
+            }
+            AZStd::vector<float> output(outputSize);
+            outputs[i] = output;
+        }
+        m_outputs = outputs;
+    }
+
+    void Model::Run(AZStd::vector<AZStd::vector<float>>& inputs)
+    {
+        m_timer.Stamp(); // Start timing of inference.
+
+        // Tensor creation is lightweight, and a tensor is just a wrapper around the memory owned by the vector passed in as data during creation.
+        // As such, creating input and output tensors in each run call does not adversely affect performance.
+        AZStd::vector<Ort::Value> inputTensors;
+        for (int i = 0; i < m_inputCount; i++)
+        {
+            Ort::Value inputTensor =
+                Ort::Value::CreateTensor<float>(m_memoryInfo, inputs[i].data(), inputs[i].size(), m_inputShapes[i].data(), m_inputShapes[i].size());
+            inputTensors.push_back(std::move(inputTensor));
+        }
+
+        AZStd::vector<Ort::Value> outputTensors;
+        for (int i = 0; i < m_outputCount; i++)
+        {
+            Ort::Value outputTensor =
+                Ort::Value::CreateTensor<float>(m_memoryInfo, m_outputs[i].data(), m_outputs[i].size(), m_outputShapes[i].data(), m_outputShapes[i].size());
+            outputTensors.push_back(std::move(outputTensor));
```

Comment:
nitpick: AZStd::move should be used instead of std::move.

Reply:
Changed